api.c

```diff
@@ -107,9 +111,9 @@ static int api_handle_req_add_hostfwd(Slirp *slirp, int fd, struct api_ctx *ctx,
         exit(EXIT_FAILURE);
     }
     fwd->is_udp = -1; /* TODO: support SCTP */
-    if (strcmp(proto_s, "udp") == 0) {
+    if (strncmp(proto_s, "udp", 3) == 0) {
         fwd->is_udp = 1;
-    } else if (strcmp(proto_s, "tcp") == 0) {
+    } else if (strncmp(proto_s, "tcp", 3) == 0) {
```

Comment:
We should reject a invalid input like “tcp5”

Reply:
It is there at line 170.